ReSharper.FSharp/src/FSharp.Psi.Intentions/src/QuickFixes/UpdateSignatureFileFix.fs

```diff
@@ -0,0 +1,73 @@
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open JetBrains.ReSharper.Psi.ExtensionsAPI.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Resources.Shell
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Services.Util
+
+type UpdateSignatureFileFix(binding: IBinding) =
+    inherit FSharpQuickFixBase()
+
+    let moduleDecl =
+        if isNull binding then None else
+        let letBinding = LetBindingsDeclarationNavigator.GetByBinding(binding)
+        if isNull letBinding then None else
+        let moduleDecl = ModuleDeclarationNavigator.GetByMember(letBinding)
+        if isNull moduleDecl then None else Some (letBinding, moduleDecl)
+    
+    new (error: SignatureFileMismatchError) =
+        UpdateSignatureFileFix(BindingNavigator.GetByHeadPattern(error.Pattern))
+
+    new (warning: ArgumentNameMismatchWarning) =
+        // The name is wrong in this case
+        // The parameter pattern is two levels higher
+        match warning.Pattern.Parent with
+        | :? ITypedPat as tp ->
+            match tp.Parent with
+            | :? IParenPat as pat ->
+                UpdateSignatureFileFix(BindingNavigator.GetByParameterPattern(pat))
+            | _ -> UpdateSignatureFileFix(Unchecked.defaultof<IBinding>)
+        | _ -> UpdateSignatureFileFix(Unchecked.defaultof<IBinding>)
+    
+    override this.Text = "Update signature file"
+    
+    override this.IsAvailable _ =
+        Option.isSome moduleDecl
```

Comment:
We should also check that types used in a new signature are available in the signature file:
```fsharp
module M

type T() = class end

let f (_: T) = ()
```
```fsharp
module M

val f: int -> unit
```